Tasks/AzureRmWebAppDeployment/XMLSubstituter.ps1

```diff
@@ -0,0 +1,74 @@
+﻿Param(
+  [Parameter(mandatory=$true)][string]$xmlFilePath,
+  [Parameter(mandatory=$true)][string]$tag,
+  [string]$conditionalAttributeKey,
+  [string]$conditionalAttributeValue,
+  [Parameter(mandatory=$true)][string]$targetKey,
+  [Parameter(mandatory=$true)][string]$targetValue
+)
+
+Write-Verbose "xmlFile = $xmlFilePath" -Verbose
+Write-Verbose "Tag = $tag" -Verbose
+Write-Verbose "conditionalAttribute = $conditionalAttributeKey" -Verbose
+Write-Verbose "conditionalValue = $conditionalAttributeValue" -Verbose
+Write-Verbose "targetKey = $targetKey" -Verbose
+Write-Verbose "targetValue = $targetValue" -Verbose
+
+
+if( $conditionalAttribute -eq $null -xor $conditionalValue -eq $null) {
```

Comment:
use String::IsNullOrEmpty($param) which is recommended in powershell for null checks